lib/tests/test_rate_map.c

```diff
@@ -177,6 +177,18 @@ test_translate_position_and_recomb_mass(void)
     /* inside recombination interval */
     CU_ASSERT_EQUAL(rate_map_position_to_mass(&map, 8), 18);
 
+    /* max valid position */
+    CU_ASSERT_EQUAL(rate_map_position_to_mass(&map, 20), 25);
+
+    /* NaNnoying cases */
+#ifdef NDEBUG // test via `meson -Db_ndebug=true ndebug_build`
```

Comment:
I don't want to have conditional compiles in the test suite, everything is already quite complicated and this seems gratuitous.

Reply:
Good instinct to discourage #ifdef'd code sprinkled inside code. I imagine in open-source that can really encourage excessive #ifdef use. And with git, it seems like lots of traditional uses of #ifdef in development are better handled with git branches.

I'll replace all these NaNnoying cases and with just that one line testing for finite -1 without ifdef.

This relates to a deeper question which I think is not really resolved yet which is what are the desirable uses of NDEBUG conditioned assert.h assert() calls in msprime/tskit. Not sure what's the best forum or time to dig deeper into this question.